src/main/java/com/fasterxml/jackson/databind/deser/std/StringDeserializer.java

```diff
@@ -36,6 +41,81 @@ public Object getEmptyValue(DeserializationContext ctxt) throws JsonMappingExcep
 
     @Override
     public String deserialize(JsonParser p, DeserializationContext ctxt) throws IOException
+    {
+        // disabled, execute default serialization
+        if (!ctxt.isEnabled(DeserializationFeature.ACCEPT_SUB_JSON_AS_STRING)) {
+            return defaultDeserialize(p, ctxt);
+        }
+
+        JsonToken currentToken = p.getCurrentToken();
+
+        // not a JSON substring, execute default serialization
+        if (currentToken != JsonToken.START_OBJECT && currentToken != JsonToken.START_ARRAY) {
+            return defaultDeserialize(p, ctxt);
+        }
+
+        StringBuilder builder = new StringBuilder();
```

Comment:
Ugh. This is madness... are we trying to re-construct JSON BACK from decoded JSON.

No, I don't think this is something to add.